server/planning/commands/delete_spiked_items.py

```diff
@@ -30,123 +39,125 @@ class DeleteSpikedItems(Command):
+        return True, [historic + past]
 
-    def _delete_spiked_planning(self, expiry_datetime):
-        logger.info("{} Starting to delete spiked planning items".format(self.log_msg))
-        planning_service = get_resource_service("planning")
+    return False, []
 
-        # Obtain the full list of Planning items that we're to process first
-        # As subsequent queries will change the list of returnd items
-        plans = dict()
-        for items in planning_service.get_expired_items(expiry_datetime, spiked_planning_only=True):
-            plans.update({item[ID_FIELD]: item for item in items})
 
-        plans_deleted = set()
-        assignments_deleted = set()
-        assignments_to_delete = []
+async def delete_spiked_planning(expiry_datetime):
+    log_msg = log_msg_context.get()
+    logger.info(f"{log_msg} Starting to delete spiked planning items")
+    planning_service = PlanningAsyncService()
 
-        for plan_id, plan in plans.items():
-            for coverage in plan.get("coverages") or []:
-                assignment_id = (coverage.get("assigned_to") or {}).get("assignment_id")
-                if assignment_id:
-                    assignments_to_delete.append(assignment_id)
+    # Obtain the full list of Planning items that we're to process first
+    # As subsequent queries will change the list of returnd items
+    plans = dict()
+    async for items in planning_service.get_expired_items(expiry_datetime, spiked_planning_only=True):
+        plans.update({item[ID_FIELD]: item for item in items})
 
-            # Now, delete the planning item
-            planning_service.delete_action(lookup={"_id": plan_id})
-            plans_deleted.add(plan_id)
+    plans_deleted = set()
+    assignments_deleted = set()
+    assignments_to_delete = []
 
-        # Delete assignments
-        assignment_service = get_resource_service("assignments")
-        for assign_id in assignments_to_delete:
-            assignment_service.delete(lookup={"_id": assign_id})
-            assignments_deleted.add(assign_id)
+    for plan_id, plan in plans.items():
+        for coverage in plan.get("coverages") or []:
+            assignment_id = (coverage.get("assigned_to") or {}).get("assignment_id")
+            if assignment_id:
+                assignments_to_delete.append(assignment_id)
 
-        logger.info(
-            "{} {} Assignments deleted: {}".format(self.log_msg, len(assignments_deleted), list(assignments_deleted))
-        )
-        logger.info("{} {} Planning items deleted: {}".format(self.log_msg, len(plans_deleted), list(plans_deleted)))
+        # Now, delete the planning item
+        await planning_service.delete_action(lookup={"_id": plan_id})
+        plans_deleted.add(plan_id)
 
+    # Delete assignments
+    assignment_service = AssingmentsAsyncService()
+    for assign_id in assignments_to_delete:
+        await assignment_service.delete(lookup={"_id": assign_id})
```

Comment:
And once more, use the `delete_many` method. Otherwise if you want to delete a single item you can:

```python
assignment = await assignment_service.find_by_id(assign_id)
await assignment_service.delete(assignment)
```